dbms/src/Storages/Page/V2/VersionSet/PageEntriesVersionSetWithDelta.h

```diff
@@ -77,6 +78,29 @@ class PageEntriesVersionSetWithDelta
 
     size_t sizeUnlocked() const;
 
+    bool tryCompact()
+    {
+        const auto release_idx = last_released_snapshot_index.load();
+        const auto last_try_idx = last_try_compact_index.load();
+        if (release_idx <= last_try_idx)
+        {
+            return false;
+        }
+
+        // some new snapshot are released, let's try
+        // compact the versions.
+        last_try_compact_index.store(release_idx);
+
+        // compact version list with the latest snapshot.
+        // do NOT increase the index by this snapshot or it will
+        // cause inf loop
+        auto snap = getSnapshot("ps-mem-compact", nullptr);
+        compactOnDeltaRelease(snap->view.getSharedTailVersion());
```

Comment:
`compactOnDeltaRelease` I think this method need a new name.

Reply:
Change to `compactUntil`?